molgenis-api/src/main/java/org/molgenis/api/convert/UrlParameterUtils.java

```diff
@@ -0,0 +1,152 @@
+    }
+    String queryParam = query.isPresent() ? queryToUrlParam(query.get()) : "";
+    if (!queryParam.isEmpty()) {
+      params.put(QUERY, queryParam);
+    }
+
+    params.put(PAGE, Integer.toString(number));
+    params.put(SIZE, Integer.toString(size));
+
+    return params;
+  }
+
+  public static String createUrlParamString(
+      Selection filter, Selection expand, Sort sort, int size, int number, Optional<Query> query) {
+    Map<String, String> paramMap = createUrlParams(filter, expand, sort, size, number, query);
+    return paramMap.entrySet().stream()
+        .filter(entry -> !entry.getValue().isEmpty())
+        .map(entry -> entry.getKey() + "=" + entry.getValue())
+        .collect(Collectors.joining("&"));
+  }
+
+  static String queryToUrlParam(Query query) {
+    return queryToUrlParam(query, false);
+  }
+
+  static String queryToUrlParam(Query query, boolean hasParent) {
+    String param;
+    Operator queryOperator = query.getOperator();
+    switch (queryOperator) {
+      case EQUALS:
+        param = query.getItem() + "==" + query.getStringValue();
+        break;
+      case NOT_EQUALS:
+        param = query.getItem() + "!=" + query.getStringValue();
+        break;
+      case IN:
+        param =
+            query.getItem()
+                + "=in="
+                + query.getStringListValue().stream().collect(Collectors.joining(","));
+        break;
+      case NOT_IN:
+        param =
+            query.getItem()
+                + "=out="
+                + query.getStringListValue().stream().collect(Collectors.joining(","));
+        break;
+      case LESS_THAN:
+        param = query.getItem() + "<" + query.getStringValue();
+        break;
+      case LESS_THAN_OR_EQUAL_TO:
+        param = query.getItem() + "<=" + query.getStringValue();
+        break;
+      case GREATER_THAN:
+        param = query.getItem() + ">" + query.getStringValue();
+        break;
+      case GREATER_THAN_OR_EQUAL_TO:
+        param = query.getItem() + ">=" + query.getStringValue();
+        break;
+      case CONTAINS:
+        param = query.getItem() + "=like=" + query.getStringValue();
+        break;
+      case MATCHES:
+        param = query.getItem() + "=q=" + query.getStringValue();
+        break;
+      case AND:
+        String andQuery =
+            query.getQueryListValue().stream()
+                .map(subQuery -> queryToUrlParam(subQuery, true))
+                .collect(Collectors.joining(";"));
+        param = hasParent ? "(" + andQuery + ")" : andQuery;
+        break;
+      case OR:
+        String orQuery =
+            query.getQueryListValue().stream()
+                .map(subQuery -> queryToUrlParam(subQuery, true))
+                .collect(Collectors.joining(","));
+        param = hasParent ? "(" + orQuery + ")" : orQuery;
+        break;
+      default:
+        throw new UnexpectedEnumException(queryOperator);
+    }
+    return param;
+  }
+
+  static String sortToUrlParam(Sort sort) {
```

Comment:
Move to SortConverter so that we 'from' and 'to' converters in the same class? Same for *ToUrlParam converters.